src/build/browser/browser-static/bootstrap.js

```diff
@@ -134,11 +92,39 @@ function bootstrap(initialImport, target) {
 
 	var loaded = false;
 	w._continueLoad = function() {
-		if (!loaded) {
-			loaded = true;
-			var el = d.createElement('script');
-			el.src = target + '.js';
-			d.getElementsByTagName('head')[0].appendChild(el);
+		var doThings = function() {
+			if (!loaded) {
+				loaded = true;
+				// Include the game code
+				var el = d.createElement('script');
+				el.src = target + '.js';
+				d.getElementsByTagName('head')[0].appendChild(el);
+			}
+		};
+
+		var _continueLoadDefer;
+
+		w.addEventListener('message', function(event) {
+			if (event.data === 'partialLoadContinue') {
+				if (_continueLoadDefer) {
+					_continueLoadDefer.resolve();
+					_continueLoadDefer = undefined;
+				}
+			}
+		});
+		// Preload suggestions and then tell the parent bootstrapping is complete
+		jsio.__env.preloadModules(function() {
+			w.parent.postMessage('bootstrapping', '*');
+		});
+
+		var partialLoadKey = jsio.__env.getNamespace('partialLoad');
+		if (localStorage && localStorage.getItem(partialLoadKey)) {
+			localStorage.removeItem(partialLoadKey);
+
+			_continueLoadDefer = Promise.defer();
```

Comment:
actually, i was wrong, it's still pretty bad though -- adding in chrome for android you get:

chrome android: 13.77%
android browser with promises: 1.13%
total android: 13.77 + 1.13 + 2.61 + 1.48 + 0.75
`(13.77+1.13) / (13.77 + 1.13 + 2.61 + 1.48 + 0.75) == ~75%`

so according to that particular website, about a 1 in 4 chance it fails on an arbitrary android device? i'd *maybe* buy the argument that devkit games only run on chrome for android and that we don't support the old android webkit, but chrome doesn't ship with android (oem's can choose to ship it i think) and I'm not sure which devices we'd actually expect to have chrome these days. Until Chrome is default on android or we have usage numbers of 90-95%+ chrome on android, I don't think it's wise to drop support for older browsers, and I'd definitely prefer some kind of nice you-need-to-upgrade message, not just have it throw an exception on load with a white screen.